tests/components/radarr/__init__.py

```diff
@@ -1 +1,204 @@
+        headers={"Content-Type": CONTENT_TYPE_JSON},
+    )
+
+    aioclient_mock.get(
+        f"{url}/api/v3/command",
+        text=load_fixture("radarr/command.json"),
+        headers={"Content-Type": CONTENT_TYPE_JSON},
+    )
+
+    if windows:
+        aioclient_mock.get(
+            f"{url}/api/v3/rootfolder",
+            text=load_fixture("radarr/rootfolder-windows.json"),
+            headers={"Content-Type": CONTENT_TYPE_JSON},
+        )
+    else:
+        aioclient_mock.get(
+            f"{url}/api/v3/rootfolder",
+            text=load_fixture("radarr/rootfolder-linux.json"),
+            headers={"Content-Type": CONTENT_TYPE_JSON},
+        )
+
+    aioclient_mock.get(
+        f"{url}/api/v3/movie",
+        text=load_fixture("radarr/movie.json"),
+        headers={"Content-Type": CONTENT_TYPE_JSON},
+    )
+
+
+def mock_connection_error(
+    aioclient_mock: AiohttpClientMocker,
+    url: str = URL,
+) -> None:
+    """Mock radarr connection errors."""
+    aioclient_mock.get(f"{url}/api/v3/system/status", exc=ClientError)
+
+
+def mock_connection_invalid_auth(
+    aioclient_mock: AiohttpClientMocker,
+    url: str = URL,
+) -> None:
+    """Mock radarr invalid auth errors."""
+    aioclient_mock.get(f"{url}/api/v3/system/status", status=HTTPStatus.UNAUTHORIZED)
+    aioclient_mock.get(f"{url}/api/v3/command", status=HTTPStatus.UNAUTHORIZED)
+    aioclient_mock.get(f"{url}/api/v3/movie", status=HTTPStatus.UNAUTHORIZED)
+    aioclient_mock.get(f"{url}/api/v3/rootfolder", status=HTTPStatus.UNAUTHORIZED)
+
+
+def mock_connection_server_error(
+    aioclient_mock: AiohttpClientMocker,
+    url: str = URL,
+) -> None:
+    """Mock radarr server errors."""
+    aioclient_mock.get(
+        f"{url}/api/v3/system/status", status=HTTPStatus.INTERNAL_SERVER_ERROR
+    )
+    aioclient_mock.get(f"{url}/api/v3/command", status=HTTPStatus.INTERNAL_SERVER_ERROR)
+    aioclient_mock.get(f"{url}/api/v3/movie", status=HTTPStatus.INTERNAL_SERVER_ERROR)
+    aioclient_mock.get(
+        f"{url}/api/v3/rootfolder", status=HTTPStatus.INTERNAL_SERVER_ERROR
+    )
+
+
+async def setup_integration(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    url: str = URL,
+    api_key: str = API_KEY,
+    unique_id: str = None,
+    skip_entry_setup: bool = False,
+    connection_error: bool = False,
+    invalid_auth: bool = False,
+    windows: bool = False,
+) -> MockConfigEntry:
+    """Set up the radarr integration in Home Assistant."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=unique_id,
+        data={
+            CONF_URL: url,
+            CONF_API_KEY: api_key,
+            CONF_VERIFY_SSL: False,
+        },
+    )
+
+    entry.add_to_hass(hass)
+
+    mock_connection(
+        aioclient_mock,
+        url=url,
+        error=connection_error,
+        invalid_auth=invalid_auth,
+        windows=windows,
+    )
+
+    if not skip_entry_setup:
```

Comment:
Don't have conditions in tests. Use pytest fixtures instead to compose the test setup.